### PR TITLE
comments entity,interface and dto update

### PR DESCRIPTION
**Pull request by DLiSWE**
Comments entity, interface and dto typed.
Post relationships included in both comments and user entity and interface.

### CODE REVIEW

File: src/dtos/comments.dto.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    @IsNumber()
+    public user_id: number;
+
+    @IsString()
+    public media: string;
+
+    @IsString()
+    public context: string;
+
+    @IsDate()
```
Comment (raylu511) [inline src/dtos/comments.dto.ts:17]: We don't need created_at and updated_at in DTO.
DTOs define the incoming request object
Created_at and updated_at are handled by typeorm and not the client

File: src/controllers/index.controller.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ class IndexController {
   public index = (_req: Request, res: Response, next: NextFunction) => {
     try {
       res.sendStatus(200);
+
```
Comment (raylu511) [inline src/controllers/index.controller.ts:7]: Nit: Remove newline

File: src/dtos/comments.dto.ts (hunk 1 above)
Comment (emir4ik-code) [reply src/dtos/comments.dto.ts:17]: Is there a need to use Class Validator if we're using typescript anyway?

File: src/dtos/comments.dto.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    readonly user: number;
+    readonly post: number;
+    readonly media?: string;
+    readonly content: string;
+    readonly createdAt: Date;
+    readonly updatedAt: Date;
+}
+
+// export type CreateCommentDto = Omit<BaseCommentDto, 'id'>
+// export type UpdateCommentDto = Omit<BaseCommentDto, 'created_at'>
```
Comment (emir4ik-code) [inline src/dtos/comments.dto.ts:12]: we can uncomment this.

File: src/entities/comments.entity.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    @JoinColumn()
+    user: UserEntity;
+
+    @Column()
+    media: string;
+
+    @Column()
+    content: string
+
+    @Column()
```
Comment (emir4ik-code) [inline src/entities/comments.entity.ts:31]: We don't need Column decorator, when we use special columns

File: src/entities/comments.entity.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    @CreateDateColumn()
+    createdAt: Date
+
+    @Column()
```
Comment (emir4ik-code) [inline src/entities/comments.entity.ts:35]: Same here

Comment (DLiSWE) [reply src/entities/comments.entity.ts:35]: Ok

**Review by raylu511 — APPROVED** [review]

File: src/entities/comments.entity.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    // @ManyToOne(()=>PostEntity, post=> post.comments)
+    // @JoinColumn()
+    // post: PostEntity;
+
+    @ManyToOne(()=>UserEntity, user => user.id)
+    @JoinColumn()
+    user: UserEntity;
+
+    @Column()
+    media: string;
```
Comment (emir4ik-code) [inline src/entities/comments.entity.ts:26]: should this be nullable?

Comment (raylu511) [reply src/entities/comments.entity.ts:26]: Good catch.

**Review by emir4ik-code — APPROVED** [review]

**Pull request merged**